app/views/docs/certificates.phtml

```diff
@@ -9,12 +9,34 @@
 
 <h2>Debugging</h2>
 
-<p>In case you&#39;re still struggling with getting your certificate to work, check the Appwrite certificates worker log. You can do that with the following command:</p>
+<p>If you're still struggling with your certificates, check the Appwrite certificates worker log. You can do that with the following command:</p>
 
 <div class="ide margin-bottom" data-lang="bash" data-lang-label="Bash">
     <pre class="line-numbers"><code class="prism language-bash" data-prism>docker-compose logs appwrite-worker-certificates</code></pre>
 </div>
 
+<h2>Generation Cycle</h2>
+
+<p>Appwrite auto-generates a certificate for your main domain when you first visit it. If your browser shows an insecure connection warning, you must proceed to trigger certificate generation. The domain in environment variable _APP_DOMAIN is considered your main domain. If you didn't set this variable, the first domain you visit would be marked as the main domain for your Appwrite instance. Appwrite follows this concept of the main domain to prevent generating certificates for domains you don't allow. Keep in mind that you can add a domain as a custom domain in your project settings to enable certificate generation for a any domain.</p>
+
+<p>Certificate renewal is done as a part of the Appwrite maintenance task. Unless modified with environment variable _APP_MAINTENANCE_INTERVAL, this task runs every 24 hours. During this task, Appwrite looks for certificates due to renewal and renews them. One maintenance cycle only attempts to renew up to 200 certificates to respect Let'sEncrypt API limit. Every Let's Encrypt certificate is valid for 90 days, but Appwrite starts to renew them 30 days before the expiration.</p>
+
+<p>There are many reasons why certificate generation could fail, such as incorrect DNS settings, hitting Let's Encrypt API limit, or stopped Appwrite HTTP server. Every certificate generation failure is attempted again in the next renewal cycle, but only up to 5 attempts. You can use manual generation if you need to run certificate generation for a domain after passing the attempts limit.</p>
+
+<h2>Manual Generation</h2>
+
+<p>Appwrite generates and renews certificates automatically, so manual generation is not required. A manual generation can be useful when you hit the API limit and don't want to wait for the next maintenance cycle to renew the certificate. Use the following command to generate a certificate for your main domain:</p>
```

Comment:
```suggestion
<p>Since Appwrite generates and renews certificates automatically, a manual generation is seldom required. A manual generation can be useful when you hit the API limit and don't want to wait for the next maintenance cycle to renew the certificate. Use the following command to generate a certificate for your main domain:</p>
```